File: ce/outlook-app/snippets/visualbasic/CRMV8/businessmanagementvb/vb/createqueue1.vb
' <snippetcreatequeue1>


 ' Define some anonymous types to define the range of possible 
 ' queue property values.
 Dim IncomingEmailDeliveryMethods = New With {Key .None = 0, Key .EmailRouter = 2, Key .ForwardMailbox = 3}

 Dim IncomingEmailFilteringMethods = New With {Key .AllEmailMessages = 0, Key .EmailMessagesInResponseToCrmEmail = 1, _
                                               Key .EmailMessagesFromCrmLeadsContactsAndAccounts = 2}

 Dim OutgoingEmailDeliveryMethods = New With {Key .None = 0, Key .EmailRouter = 2}

 ' Create a queue instance and set its property values.
 Dim newQueue As New Queue() With {.Name = "Example Queue.", .Description = "This is an example queue.", _
                     .IncomingEmailDeliveryMethod = New OptionSetValue(IncomingEmailDeliveryMethods.None), _
                     .IncomingEmailFilteringMethod = New OptionSetValue(IncomingEmailFilteringMethods.AllEmailMessages), _
                                   .OutgoingEmailDeliveryMethod = New OptionSetValue(OutgoingEmailDeliveryMethods.None)}

 ' Create a new queue instance.
 _queueId = _serviceProxy.Create(newQueue)

' </snippetcreatequeue1>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/businessmanagement/cs/createqueue1.cs
// <snippetcreatequeue1>


 // Define some anonymous types to define the range of possible 
 // queue property values.
 var IncomingEmailDeliveryMethods = new
 {
     None = 0,
     EmailRouter = 2,
     ForwardMailbox = 3
 };

 var IncomingEmailFilteringMethods = new
 {
     AllEmailMessages = 0,
     EmailMessagesInResponseToCrmEmail = 1,
     EmailMessagesFromCrmLeadsContactsAndAccounts = 2
 };

 var OutgoingEmailDeliveryMethods = new
 {
     None = 0,
     EmailRouter = 2
 };

 var QueueViewType = new
 {
     Public = 0,
     Private = 1
 };
 // Create a queue instance and set its property values.
 Queue newQueue = new Queue()
 {
     Name = "Example Queue.",
     Description = "This is an example queue.",
     IncomingEmailDeliveryMethod = new OptionSetValue(
         IncomingEmailDeliveryMethods.None),
     IncomingEmailFilteringMethod = new OptionSetValue(
         IncomingEmailFilteringMethods.AllEmailMessages),
     OutgoingEmailDeliveryMethod = new OptionSetValue(
         OutgoingEmailDeliveryMethods.None),
     QueueViewType = new OptionSetValue(
         QueueViewType.Private)
 };

 // Create a new queue instance.
 _queueId = _serviceProxy.Create(newQueue);

// </snippetcreatequeue1>